compiler/modules/nback/equal.bal

```diff
@@ -59,20 +59,19 @@ final RuntimeFunction decimalExactEqFunction = {
 type CmpEqOp "ne"|"eq";
 
 function buildEquality(llvm:Builder builder, Scaffold scaffold, bir:EqualityInsn insn) returns BuildError? {
-    var [lhsRepr, lhsValue] = check buildReprValue(builder, scaffold, insn.operands[0]);
-    var [rhsRepr, rhsValue] = check buildReprValue(builder, scaffold, insn.operands[1]);
+    var [lhs, rhs] = insn.operands;
+    var [lhsRepr, lhsValue] = check buildReprValue(builder, scaffold, lhs);
+    var [rhsRepr, rhsValue] = check buildReprValue(builder, scaffold, rhs);
     CmpEqOp op = insn.op[0] == "!" ?  "ne" : "eq"; 
     boolean exact = insn.op.length() == 3; // either "===" or "!=="
     bir:Register result = insn.result;
     match [lhsRepr.base, rhsRepr.base] {
         [BASE_REPR_TAGGED, BASE_REPR_TAGGED] => {
-            if reprIsImmediate(lhsRepr) || reprIsImmediate(rhsRepr) {
+            // JBUG cast to boolean
+            if <boolean>lhsRepr.isImmediate || <boolean>rhsRepr.isImmediate || <boolean>semTypeRepr(t:intersect(lhs.semType, rhs.semType)).isImmediate {
```

Comment:
I suggest we work around this bug by adding a reprIsImmediate function that does `<boolean>repr.isImmediate`